docs/source/community/pull_requests.rst

```diff
@@ -0,0 +1,360 @@
+direction. That said, particularly in a large project like NWChemEx it can be
+almost impossible for any one person to fully understand what can break any
+given component or for any one person to comprehend the entire design and
+vision. The point being, while PRs do protect against malicious attacks, they
+are primarily meant to avoid accidental breakages. Part of this is through
+automated continuous integration pipelines (making sure the code builds and
+tests pass, etc.) and part of this is through manual code review, *i.e.*,
+having people read your code.
+
+Modern PRs are about more than just code review, they also are meant to be
+learning opportunities for both experienced and inexperienced coders. Again,
+no one person knows every aspect of a code, so when you make a PR it may turn
+out that the reviewer knows a better way to do something. This works both ways
+and reviewers often learn new techniques or discover previously unknown
+features by reviewing code.
+
+PRs are also about staking a claim. When you open a draft PR you're telling
+people that you are going to work on resolving a bug etc. This tells other
+people not to duplicate the effort and/or gives them a chance to join the
+cause.
+
+
+*****************
+PR Considerations
+*****************
+
+This section lists considerations for authors and/or reviewers of PRs.
+Following semantic versioning, we can think of each PR as addressing one of
+three use cases:
+
+#. API breaking changes
+
+   - Any change to a public-facing API which breaks previously working code.
+
+#. New features
+
+   - Introduces new code without breaking the API
+   - New alternative APIs
+   - Allows the code to compute new things
+   - Quality of life improvements (e.g., extensive optimization)
+   - Entirely new documentation sections
+
+#. Patch
+
+   - Largely modifies existing code
+   - Bug fixes, typos, documentation tweaks
+   - Performance optimizations spanning a couple lines of code
+
+In general the PR considerations are going to be highly tied to what the PR
+wants to accomplish, the following subsections group considerations by use cases
+and assume archetypal PRs. In other words, not all points are going to be
+relevant to every PR. If they're not relevant for a specific PR they should be
+ignored.
+
+General Considerations
+======================
+
+#. Making a PR should be easy.
+
+   - If it's too hard to make a PR people won't do it.
+   - Setting the bar too high discourages new contributors.
+
+#. Should be possible to open the PR early
+
+   - Channel for discussion between reviewers and the PR author.
+   - Gets work on radar of reviewers and other developers.
+   - Time to hash out what is and is not in scope before time is wasted
+
+#. PR lifetimes should be short.
+
+   - The code base moves fast, PRs get stale quickly.
+   - Shoot for merging in less than two weeks.
+   - Merge function by function if necessary.
```

Comment:
Would you track the overall functionality that you are adding in a "large" PR or an issue or......

Reply:
issue, added a link.